pkg/interactive-installer/src/utils.rs

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2024 Zededa, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use std::{fs, os::unix::prelude::FileTypeExt};
+use cursive::Cursive;
+use serde_json::Value;
+
+use crate::{
+    actions::execute,
+    error::Error,
+    state::{GlobalState, Move},
+};
+
+pub fn read_installer_json(json_file: &str) -> Value {
+    let text = std::fs::read_to_string(json_file).unwrap();
+    // Parse the string into a dynamically-typed JSON structure.
+    serde_json::from_str::<Value>(&text).unwrap()
+}
+
+pub fn get_state_mut(c: &mut Cursive) -> Result<GlobalState, Error> {
+    match c.take_user_data::<GlobalState>() {
+        Some(data) => {
+            c.set_user_data(data.clone());
+            Ok(data)
+        }
+        None => Err(Error::NoState),
+    }
+}
+
+pub fn get_block_devices() -> Option<Vec<String>> {
+    let mut vec = Vec::new();
+    let devpaths = fs::read_dir("/dev/").unwrap();
+    for path in devpaths {
+        let devname = path.unwrap().path();
+        let meta = fs::metadata(devname.clone()).unwrap();
+        let file_type = meta.file_type();
+        let name = devname.into_os_string().into_string().unwrap();
+
+        // We're interested in block devices, so skip non-block devices
+        if !name.contains("sd") && !name.contains("hd") && !name.contains("nvme") {
```

Comment:
not a very reliable check how about 'mmc'? maybe better to scan /sys/class/block ?

Reply:
Thank you, @rucoder, I appreciated it. I will make the changes.